lucene/core/src/java/org/apache/lucene/search/RegexpQuery.java

```diff
@@ -96,16 +96,46 @@ public RegexpQuery(Term term, int flags, int maxDeterminizedStates) {
    * Constructs a query for terms matching <code>term</code>.
    * 
    * @param term regular expression.
-   * @param flags optional RegExp features from {@link RegExp}
+   * @param maxDeterminizedStates maximum number of states that compiling the
+   * @param syntax_flags optional RegExp syntax features from {@link RegExp}
+   *  automaton for the regexp can result in.  Set higher to allow more complex
+   *  queries and lower to prevent memory exhaustion.
+   * @param match_flags boolean 'or' of match behavior options such as case insensitivity
+   */
+  public RegexpQuery(Term term,  int maxDeterminizedStates, int syntax_flags, int match_flags) {
```

Comment:
I'd keep maxDeterminizedStates last so that all constructors take parameters in the same order.